### PR TITLE
Disallow use of octal notation of IP addresses

### PR DESCRIPTION
**Pull request by AniruddhaKanhere**
Description
-----------
This PR bans the use of octal notations of IP addresses (of the format `0[0-9]+`) as per [RFC 6943 section 3.1.1](https://datatracker.ietf.org/doc/html/rfc6943#section-3.1.1).

Test Steps
-----------

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by htibosch — APPROVED** [review]
At the end of the function, I would set/clear `pvDestination` unconditionally.

It might contain an earlier IP-address, or a random value of an automatic variable.

If the user forgets to test the result, it is better to work with 0.0.0.0 rather than an existing IP-address.

The function `FreeRTOS_inet_pton6()` also clears/sets the pointer unconditionally.

Otherwise I think the PR is all good!

File: FreeRTOS_Sockets.c [hunk 1]
```diff
@@ -2450,6 +2450,8 @@ BaseType_t FreeRTOS_inet_pton4( const char * pcSource,
     const char * pcIPAddress = pcSource;
     const void * pvCopySource;
 
+    ( void ) memset( pvDestination, 0, sizeof( ulReturn ) );
+
```
Comment (htibosch) [inline FreeRTOS_Sockets.c:2454]: Yes, that also works!
Thanks

**Review by htibosch — DISMISSED** [review]

File: test/unit-test/FreeRTOS_Sockets/FreeRTOS_Sockets_utest.c [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    ulIPAddress = 0xABABABAB;
+    xResult = FreeRTOS_inet_pton4( pucValidString5, &ulIPAddress );
+    TEST_ASSERT_EQUAL( pdPASS, xResult );
+    TEST_ASSERT_EQUAL_MESSAGE( ulValidResponse5, ulIPAddress, "Could not convert string 5 correctly." );
+
+    /* Invalid test cases. */
+    ulIPAddress = 0xABABABAB;
+    xResult = FreeRTOS_inet_pton4( pucInvalidString1, &ulIPAddress );
+    TEST_ASSERT_EQUAL( pdFAIL, xResult );
+    TEST_ASSERT_EQUAL_MESSAGE( ulInValidResponse, ulIPAddress, "Could not convert string 5 correctly." );
```
Comment (aggarg) [inline test/unit-test/FreeRTOS_Sockets/FreeRTOS_Sockets_utest.c:117]:
```suggestion
    TEST_ASSERT_EQUAL_MESSAGE( ulInValidResponse, ulIPAddress, "Could not convert Invalid String 1 correctly." );
```

Comment (AniruddhaKanhere) [reply test/unit-test/FreeRTOS_Sockets/FreeRTOS_Sockets_utest.c]: Updated, thanks.

**Review by aggarg — APPROVED** [review]

**Pull request merged**